### PR TITLE
[WIP] checklog-odoo improvements

### PR DESCRIPTION
**Pull request by baimont**
https://github.com/acsone/checklog-odoo/issues/1

### CODE REVIEW

File: tests/test_checklog.py [hunk 1]
```diff
@@ -3,28 +3,25 @@
...
 
-import os
-import unittest
+from pathlib import Path
 
 from click.testing import CliRunner
 
 from checklog_odoo.cli.checklog import checklog_odoo
 
-DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
```
Comment (sbidoul) [inline tests/test_checklog.py]: I'd keep this constant. `DATA_DIR = Path(__file__).parent / "data"`, to look.for test data in the  `data` directory, relative to the `tests` dir..

Comment (sbidoul) [reply tests/test_checklog.py]: @baimont this could be the cause of the test errors?

**Comment by sbidoul** [comment]
A few quick comments 

- do we need `tests/data/odoo/*` ?
- in the LICENSE and other files, the copyright is ACSONE only (but you remain an author, of course, and are credited for that in the project metadata)
- remove `__about__.py` and use `hatch-vcs` for the version
- there is no need for a test class, you can use simple functions in `test_checklog.py`

File: tests/test_checklog.py (hunk 1 above)
Comment (baimont) [reply tests/test_checklog.py]: Putting it back, no more error using pytest in local but still there in github action

**Comment by baimont** [comment]
> A few quick comments
> 
> * remove `__about__.py` and use `hatch-vcs` for the version

@sbidoul trying to use hatch-vcs makes the installation impossible

https://github.com/acsone/checklog-odoo/pull/2/commits/10f1b02daea8aa86e05f4adf91493a8ec035874b

❯ pip install -e .
Obtaining file:///home/benoit/Projects/checklog-odoo
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing editable metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):
        File "/home/benoit/.virtualenvs/checklog-odoo/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/benoit/.virtualenvs/checklog-odoo/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/benoit/.virtualenvs/checklog-odoo/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 176, in prepare_metadata_for_build_editable
          whl_basename = build_hook(metadata_directory, config_settings)
        File "/tmp/pip-build-env-_vcf5t80/overlay/lib/python3.10/site-packages/hatchling/build.py", line 78, in build_editable
          return os.path.basename(next(builder.build(wheel_directory, ['editable'])))
        File "/tmp/pip-build-env-_vcf5t80/overlay/lib/python3.10/site-packages/hatchling/builders/plugin/interface.py", line 93, in build
          self.metadata.validate_fields()
        File "/tmp/pip-build-env-_vcf5t80/overlay/lib/python3.10/site-packages/hatchling/metadata/core.py", line 243, in validate_fields
          _ = self.version
        File "/tmp/pip-build-env-_vcf5t80/overlay/lib/python3.10/site-packages/hatchling/metadata/core.py", line 128, in version
          self._version = self._get_version()
        File "/tmp/pip-build-env-_vcf5t80/overlay/lib/python3.10/site-packages/hatchling/metadata/core.py", line 226, in _get_version
          version = self.hatch.version.cached
        File "/tmp/pip-build-env-_vcf5t80/overlay/lib/python3.10/site-packages/hatchling/metadata/core.py", line 1415, in cached
          raise type(e)(message) from None
      packaging.version.InvalidVersion: Error getting the version from source `vcs`: Invalid version: '1c63fc2af2c78c80c3eaa19541c508144b11bf1e'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.


Any idea?

**Comment by sbidoul** [comment]
That's because there was a tag with a name that hatch-vcs does not support. I removed it.

File: tests/test_checklog.py (hunk 1 above)
Comment (sbidoul) [reply tests/test_checklog.py]: I can still reproduce locally, both with pytest directly, and tox.

**Comment by baimont** [comment]
@sbidoul 

I tried

`odoo -d mydb -i base --stop-after-init | checklog-odoo`

and

`odoo -d mydb -i base --stop-after-init | acsoo checklog`

which both gave me 

`Error: No Odoo log record found in input.`

And I believe this is not right, right?

**Comment by sbidoul** [comment]
@baimont that is because odoo logs on stderr and `|` reads the stdout of the process on the left.

That is one of the reaons we use `unbuffer` in CI, as it copies stderr to stdout.

So that is ok, but it may be worth mentioning in the README of this project.

File: LICENSE.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2023-present Benoit Aimont <benoit.aimont@acsone.eu>
+Copyright (c) 2023-present ACSONE SA/NV
```
Comment (sbidoul) [inline LICENSE.txt]: Since this originates in acsoo, the copyright probably starts several year before.

File: src/checklog_odoo/__init__.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# SPDX-FileCopyrightText: 2023-present ACSONE <https://acsone.eu>
+#
+# SPDX-License-Identifier: MIT
+
+from . import checklog
```
Comment (sbidoul) [inline src/checklog_odoo/__init__.py]: I think you can remove this `__init__.py`

File: README.md [hunk 4]
```diff
@@ -1,2 +1,37 @@
...
+pip install checklog-odoo
+```
+
+## Features
+
+```console
+checklog-odoo odoo.log
+unbuffer odoo -d mydb -i base --stop-after-init | checklog-odoo
+checklog-odoo --ignore "WARNING.*blah" odoo.log
+```
```
Comment (sbidoul) [inline README.md:34]: This section could be elaborated, explaining how the configuration file works, etc?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,37 @@
...
+**Table of Contents**
+
+- [Installation](#installation)
+- [Features](#features)
+- [License](#license)
+
+## Installation
+
+```console
+pip install checklog-odoo
```
Comment (sbidoul) [inline README.md]:
```suggestion
pipx install checklog-odoo
```

**Comment by baimont** [comment]
Hello @sbidoul,

Je vais avoir un peu de temps prochainement. Ce serait l'occasion de retravailler ce sujet, si tu as un moment pour reviewer avant?

File: pyproject.toml [hunk 6]
```diff
@@ -144,20 +149,14 @@ ban-relative-imports = "all"
...
-parallel = true
-omit = [
-  "src/checklog_odoo/__about__.py",
-]
+source_pkgs = ["checklog-odoo"]
 
 [tool.coverage.paths]
-checklog_odoo = ["src/checklog_odoo", "*/checklog-odoo/src/checklog_odoo"]
-tests = ["tests", "*/checklog-odoo/tests"]
+source = ["src", ".tox/*/site-packages"]
```
Comment (sbidoul) [inline pyproject.toml:118]: We should not need tox site-packages here

File: tox.ini [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+python =
+    3.7: py37
+    3.8: py38
+    3.9: py39
+    3.10: py310
+    3.11: py311
+    3.12: py312
+
+[tox]
+isolated_build = True
```
Comment (sbidoul) [inline tox.ini:11]: Can you try use_develop=True here and see if it solves the coverage config issue?

File: src/checklog_odoo/__init__.py (hunk 3 above)
Comment (baimont) [reply src/checklog_odoo/__init__.py]: no, this leads to ModuleNotFoundError: No module named 'checklog_odoo'

**Comment by baimont** [comment]
@sbidoul, @ThomasBinsfeld asks me to finish this pull request. is something blocking the merge?

**Comment by sbidoul** [comment]
> asks me to finish this pull request. is something blocking the merge?

@baimont I had not noticed you had handled the last review comments. We you do so, it's a good habit to leave a note on the PR saying it is ready to review again.

File: src/checklog_odoo/__init__.py (hunk 3 above)
Comment (sbidoul) [reply src/checklog_odoo/__init__.py]: Sorry, I meant removing the `import` in this `__init__.py`

**Comment by baimont** [comment]
> > asks me to finish this pull request. is something blocking the merge?
> 
> @baimont I had not noticed you had handled the last review comments. We you do so, it's a good habit to leave a note on the PR saying it is ready to review again.

I used 

image

which led to

image

What's the difference with a note? As I resolved all the open comments I was guessing it was enough.

File: src/checklog_odoo/__init__.py (hunk 3 above)
Comment (baimont) [reply src/checklog_odoo/__init__.py]: done

**Comment by baimont** [comment]
@sbidoul ready to review again

**Comment by baimont** [comment]
> @sbidoul ready to review again

hi there,
still the case

**Comment by baimont** [comment]
Hi @sbidoul 

Working on a new project for Liser, I realised we still used acsoo for checklog-odoo.
Probably because Arnaud used https://github.com/acsone/acsone-odoo-project-template/blob/master/src/acsoo.cfg for the initial setup.

Wouldn't it be great to merge this wonderful pr and then to update the odoo-project-template?
I know alcyon and silog use this pr so it seems to be working :-)

Just a friendly reminder, a bit more verbose this time :-p

**Comment by sbidoul** [comment]
Ok, I did some tyding up and I'll now merge. Thanks for your patience.

**Pull request merged**